Improve printing of negative exponents in PolyElement

shivamvats opened the pull request:
cc @certik

shivamvats committed:
Put negative exponents within brackets and add tests

shivamvats committed:
Use brackets for negative exponents in tests

shivamvats commented:
@certik Could you have a look at this?

certik commented:
I think this looks good, but you should also add regular tests for the negative exponents. You only have a few doctests.

certik commented on sympy/polys/rings.py:
```diff
@@ -784,7 +784,7 @@ def str(self, printer, precedence, exp_pattern, mul_symbol):
                 symbol = printer.parenthesize(symbols[i], prec_atom-1)
                 if exp != 1:
-                    if exp != int(exp):
+                    if exp != int(exp) or exp < 0:
```
This change needs regular tests somewhere in `sympy/polys`.

shivamvats committed:
A multinomial raised to negative power should raise an error.

We don't need to allow rational/negative exponents with multinomial as
the result is an infinite series in most cases. User can use ring series
functions for the same by specifying the required precision.

shivamvats committed:
Add tests for printing of powers of PolyElement monomials

shivamvats commented:
@certik

shivamvats commented:
@certik Could you have a look?

certik commented:
I think this looks great, thanks for finishing it.
